src/librustc_expand/proc_macro_server.rs

```diff
@@ -437,14 +443,8 @@ impl server::TokenStreamIter for Rustc<'_> {
                 let next = iter.cursor.next_with_joint()?;
                 Some(TokenTree::from_internal((next, self.sess, &mut iter.stack)))
             })?;
-            // HACK: The condition "dummy span + group with empty delimiter" represents an AST
-            // fragment approximately converted into a token stream. This may happen, for
-            // example, with inputs to proc macro attributes, including derives. Such "groups"
-            // need to flattened during iteration over stream's token trees.
-            // Eventually this needs to be removed in favor of keeping original token trees
-            // and not doing the roundtrip through AST.
             if let TokenTree::Group(ref group) = tree {
```

Comment:
Could you re-add the `HACK:` comment above, and link to https://github.com/rust-lang/rust/pull/73345?